src/topology/mod.rs

```diff
@@ -189,8 +207,10 @@ impl TopologyDescription {
         // Special case - If secondaries are found, by are filtered out by tag sets,
         // the topology should return any available primaries instead.
         if hosts.is_empty() && read_preference.mode == ReadMode::SecondaryPreferred {
-            let mut read_pref = read_preference.clone();
-            read_pref.mode = ReadMode::PrimaryPreferred;
+            let read_pref = ReadPreference {
+                mode: ReadMode::PrimaryPreferred,
+                ..read_preference.clone()
```

Comment:
I think so, because `read_preference` is of type `&ReadPreference` and it's not `Copy` so we can't move out of it. I'll try to remove the clone anyway, but I'm pretty sure it won't compile.

Reply:
Yes, checked it, we need the `.clone()`.